### PR TITLE
filters

### PR DESCRIPTION
**Pull request by codarose**

### CODE REVIEW

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-19)

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-19)

File: features/issues/api/use-get-issues.tsx [hunk 1]
```diff
@@ -2,32 +2,39 @@ import { useEffect } from "react";
...
   useEffect(() => {
     if (query.data?.meta.hasNextPage) {
-      queryClient.prefetchQuery(getQueryKey(page + 1), ({ signal }) =>
-        getIssues(page + 1, { signal })
+      queryClient.prefetchQuery(getQueryKey(page + 1, filters), ({ signal }) =>
+        getIssues(page + 1, filters, { signal })
       );
     }
-  }, [query.data, page, queryClient]);
+  }, [query.data, page, filters, queryClient]);
```
Comment (jkettmann) [inline features/issues/api/use-get-issues.tsx:32]: Not sure if this is creating your problem but the filters object gets created during each render. So adding it as a dependency to this useEffect causes it to get executed on each render. You could add each filter value as a separate dependency. Or use '...Object.values(filter)' as dependency which should have the same effect (if the values are always returned in the same order).

The same might be valid for the query key but I'm not sure about that. You might want to check the react-query docs about using objects as query keys

**Review by jkettmann — COMMENTED** [review]
Just one thing that stood out. If that's not the cause of your problem I can run the code on my machine in the next days

**Pull request merged**